Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/mealPlan/SelectIngredient.java

```diff
@@ -0,0 +1,139 @@
+        getIngredients();
+
+        Intent intent = getIntent();
+        mealPlan= (MealPlanDay) intent.getSerializableExtra("meal_plan_for_ingredient_add");
+
+        ingredientListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                Ingredient add_ingredient = adapter.getItem(i);
+
+                Bundle args = new Bundle();
+                args.putSerializable("meal_plan_add_ingredient", add_ingredient);
+
+                AddIngredientMPDialog addIngredientMPDialog = new AddIngredientMPDialog();
+                addIngredientMPDialog.setArguments(args);
+                addIngredientMPDialog.show(getSupportFragmentManager(), "ADD_MEAL_PLAN_INGREDIENT");
+
+            }
+        });
+
+        if (savedInstanceState == null) {
+            createTopBar();
+        }
+    }
+
+    /**
+     * Retrieves ingredients from firestore and populates a string array with the content
+     */
+    private void getIngredients() {
+        ingredientsCollection.getAll(list -> {
+            for (Ingredient ingredient : list) {
+                ingredientArrayList.add(ingredient);
+                adapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    @Override
+    public void onIngredientAdd(Ingredient meal_plan_add_ingredient) {
+        mealPlan.getIngredients().add(meal_plan_add_ingredient);
+        Intent intent1 = new Intent(getApplicationContext(), MealPlanMainScreen.class);
+        intent1.putExtra("meal_plan_after_ingredient_add", mealPlan);
+        startActivity(intent1);
+    }
+
+    @Override
+    public void onIngredientEdit(MealPlanDay meal_plan_edit_ingredient) {
+        //do nothing
+    }
+
+    /**
+     * Instantiates the top bar fragment for the recipe display menu
+     */
+    private void createTopBar() {
+        TopBar topBar = TopBar.newInstance("Select Ingredient", true, true);
+        getSupportFragmentManager().beginTransaction()
+                .setReorderingAllowed(true)
+                .replace(R.id.topBarContainerView, topBar)
+                .commit();
+    }
+
+    @Override
+    public void onAddClick() {
+        /**
+         *add a new ingredient which is not in the ingredient storage
+         */
+        Bundle args = new Bundle();
+        Ingredient new_ingredient = new Ingredient();
+        args.putSerializable("MEAL_PLAN_NEW_INGREDIENT", new_ingredient);
+        RecipeIngredientDialog addNewIngredientMPDialog = new RecipeIngredientDialog();
+        addNewIngredientMPDialog.setArguments(args);
+        addNewIngredientMPDialog.show(getSupportFragmentManager(), "ADD_NEW_MEAL_PLAN_INGREDIENT");
+
+    }
+
+
+    @Override
+    public void addRecipeIngredient(SimpleIngredient ingredient) {
+        //do nothing
+    }
+
+    @Override
+    public void editRecipeIngredient(SimpleIngredient ingredient) {
+        //do nothing
+    }
```

Comment:
Remove these if they do nothing

Reply:
I think these two are a part of an interface, so there's errors if we remove them